crates/zksync/core/src/vm/runner.rs

```diff
@@ -259,7 +263,13 @@ where
         error!("balance is 0 for {caller:?}, transaction will fail");
     }
     let max_fee_per_gas = fix_l2_gas_price(ecx.env.tx.gas_price.to_u256());
-    let gas_limit = fix_l2_gas_limit(ecx.env.tx.gas_limit.into(), max_fee_per_gas, value, balance);
+
+    let gas_limit = if paymaster_params.paymaster != Address::ZERO.to_h160() {
```

Comment:
We're using the zero value of a variable to test for absence of it. An Option is preferable here but if not we should make that explicit by defining `let use_paymaster = paymaster_params.paymaster.is_zero()`

Reply:
On it! 👍